src/main/java/ch/admin/bag/covidcertificate/api/request/CovidCertificatePersonNameDto.java

```diff
@@ -17,11 +20,24 @@
     private String givenName;
 
     public void validate() {
+        Pattern p = Pattern.compile("[!@#\\r\\n$%¶*\\\\()_:/+=|<>?{}\\[\\]~-]");
```

Comment:
I'd upgrade this attribute to a class property. The pattern compiling is counting to the method runtime, though it has no dynamic aspects and may be defined statically.

Reply:
@Armin-Isenring-Bit  But should we have any other class properties than business-properties in a Dto?

Reply:
Yeah, but in this case I'd do it as static final which should be not counted towards a dto property. But then again, I might be wrong...

Reply:
Upgraded to static final class property now